lib/docs/filters/tailwindcss/clean_html.rb

```diff
@@ -0,0 +1,63 @@
+module Docs
+  class Tailwindcss
+    class CleanHtmlFilter < Filter
+      def call
+        css('div.sticky.top-0').remove
+        css('.sr-only').remove
+        css('#sidebar').remove
+
+        css('#nav ul li').each do |node|
+          link = node.css("a").attr('href').to_s
+          if link.include? "https://"
+            node.remove()
+          end
+        end
+
+        # Remove right sidebar
+        css('#content-wrapper > div > div.hidden.xl\:text-sm.xl\:block.flex-none.w-64.pl-8.mr-8 > div > div').each do |node|
+          node.remove
+        end
+
+        # Remove advert
+        css('#__next > div.py-2.bg-gradient-to-r.from-indigo-600.to-light-blue-500.overflow-hidden').each do |node|
+          node.remove
+        end
+
+        # Remove footer prev/next navigation
+        css('#content-wrapper > div > div > div.flex.leading-6.font-medium').each do |node|
+          node.remove
+        end
+
+        css('code.language-html').each do |node|
+          node.name = 'pre';
+          node['data-language'] = 'html'
+          node.parent.name = 'div';
+          node.parent['class'] = node.parent['class'].gsub(/bg-.*?\b/, ' ');
+          node.parent.parent['class'] = node.parent.parent['class'].gsub(/bg-.*?\b/, ' ');
+        end
+
+        css('code.language-diff').each do |node|
+          node.name = 'pre';
+          node['data-language'] = 'diff'
+          node.parent.name = 'div';
+          node.parent['class'] = node.parent['class'].gsub(/bg-.*?\b/, ' ');
+          node.parent.parent['class'] = node.parent.parent['class'].gsub(/bg-.*?\b/, ' ');
+        end
+
+        css('code.language-js').each do |node|
+          node.name = 'pre';
+          node['data-language'] = 'js'
+          node.parent.name = 'div';
+          node.parent['class'] = node.parent['class'].gsub(/bg-.*?\b/, ' ');
+          node.parent.parent['class'] = node.parent.parent['class'].gsub(/bg-.*?\b/, ' ');
+        end
```

Comment:
```suggestion
        css('code').each do |node|
          node.name = 'pre';
          node.content = node.content
          node['data-language'] = node['class'][/language-(\w+)/, 1] if node['class']
          node.parent.before(node).remove
        end
```

Reply:
When I use the suggested code, I get this error when I run `thor docs:generate`
> RuntimeError: Cannot add sibling to a node with no parent

Reply:
In an attempt to use the selector, the front page has a different selector than the other pages; but I am unable to get it to work. Kindly review the latest commit and make necessary suggestions/recommendations.

```
#root page has this selector
document = doc.at_css('div.min-w-0.flex-auto.px-4.sm:px-6.xl:px-8.pt-10.pb-24.lg:pb-16')

if document.nil?
  #other pages have this selector
  document = doc.at_css('div.px-4.pt-10.pb-16')
end

@doc = document
```